pkg/cloudprovider/providers/vsphere/vclib/datastore.go

```diff
@@ -47,3 +47,23 @@ func (ds *Datastore) GetType(ctx context.Context) (string, error) {
 	}
 	return dsMo.Summary.Type, nil
 }
+
+// IsCompatibleWithStoragePolicy returns true if datastore is compatible with given storage policy else return false
+func (ds *Datastore) IsCompatibleWithStoragePolicy(ctx context.Context, storagePolicyId string) (bool, error) {
+	pbmClient, err := NewPbmClient(ctx, ds.Client())
+	if err != nil {
+		glog.Errorf("Failed to get new PbmClient Object. err: %v", err)
+		return false, err
+	}
+	datastores := []*Datastore{ds}
+	compatibleDatastore, err := pbmClient.GetCompatibleDatastores(ctx, storagePolicyId, datastores)
```

Comment:
This and the content of GetCompatibleDatastores looks very redundant and unnecessary complicated!
Called too many functions and added a lot of loop checks...
Why cannot just do the follow (or make the following another function if you like):
```
hubs := []pbmtypes.PbmPlacementHub{
		&pbmtypes.PbmPlacementHub{
			HubType: ds.Reference().Type,
			HubId:   ds.Reference().Value,
		},
	}

req := []pbmtypes.BasePbmPlacementRequirement{
		&pbmtypes.PbmPlacementCapabilityProfileRequirement{
			ProfileId: pbmtypes.PbmProfileId{
				UniqueId: storagePolicyID,
			},
		},
	}

res, err := pbmClient.CheckRequirements(ctx, hubs, nil, req)
	if err != nil {
		glog.Errorf("Error occurred for CheckRequirements call. err %+v", err)
		return false, err
	}

if len(res.CompatibleDatastores()) < 1 {
	return false, nil
} else {
	return true, nil
}
```

We only have one datastore to check.
GetCompatibleDatastores and its helper getPlacementCompatibilityResult is perfect for multiple datastores case but not for this case.
I also don't see where else GetCompatibleDatastores is called. If it's only used for here once, I suggest removing it.